### PR TITLE
Remove timeout in BlockingCollection test

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/7360
cc: @mellinoe

### CODE REVIEW

File: src/System.Collections.Concurrent/tests/BlockingCollectionTests.cs [hunk 1]
```diff
@@ -166,7 +166,7 @@ public static void TestBugFix914998()
             });
 
             int ignored;
-            Assert.Equal(1, BlockingCollection<int>.TryTakeFromAny(producerArray, out ignored, 5000));
+            Assert.Equal(1, BlockingCollection<int>.TryTakeFromAny(producerArray, out ignored));
```
Comment (stephentoub) [inline src/System.Collections.Concurrent/tests/BlockingCollectionTests.cs]: Oops, this was supposed to be passing -1. Will fix.

**Comment by mellinoe** [comment]
LGTM. If this actually does end up hanging, I can investigate why. You are probably right that it was just due to heavy load, though, since we've never seen it before.

**Comment by mellinoe** [comment]
I guess the top-line change is just due to an encoding difference or something?

**Comment by stephentoub** [comment]
Thanks.

**Pull request merged**